pkg/cloud/cloud.go

```diff
@@ -98,6 +194,305 @@ func NewCloud(region string) Cloud {
+		CopyTagsToBackups:             fileSystemOptions.CopyTagsToBackups,
+		CopyTagsToVolumes:             fileSystemOptions.CopyTagsToVolumes,
+		DailyAutomaticBackupStartTime: fileSystemOptions.DailyAutomaticBackupStartTime,
+		DeploymentType:                fileSystemOptions.DeploymentType,
+		DiskIopsConfiguration:         diskIopsConfiguration,
+		RootVolumeConfiguration:       rootVolumeConfig,
+		ThroughputCapacity:            fileSystemOptions.ThroughputCapacity,
+		WeeklyMaintenanceStartTime:    fileSystemOptions.WeeklyMaintenanceStartTime,
+	}
+
+	tags := []*fsx.Tag{
+		{
+			Key:   aws.String(AwsFsxOpenZfsDriverTagKey),
+			Value: aws.String("true"),
+		},
+		{
+			Key:   aws.String(VolumeNameTagKey),
+			Value: aws.String(fileSystemId),
+		},
+	}
+	if fileSystemOptions.Tags != nil {
+		userTags, err := c.parseTags(*fileSystemOptions.Tags)
+		if err != nil {
+			return nil, fmt.Errorf("%w. %s. %s", ErrInvalidParameter, errParseTags, err)
+		}
+		tags = append(tags, userTags...)
+	}
+
+	//Store SkipFinalBackup as a tag, so we can retrieve it during delete
+	if fileSystemOptions.SkipFinalBackup != nil {
+		tags = append(tags, &fsx.Tag{
+			Key:   aws.String(SkipFinalBackupTagKey),
+			Value: util.BoolToStringPointer(*fileSystemOptions.SkipFinalBackup),
+		})
+	}
+
+	input := fsx.CreateFileSystemInput{
+		ClientRequestToken:   aws.String(fileSystemId),
+		FileSystemType:       aws.String(DefaultFileSystemType),
+		KmsKeyId:             fileSystemOptions.KmsKeyId,
+		OpenZFSConfiguration: openZFSConfiguration,
+		SecurityGroupIds:     aws.StringSlice(fileSystemOptions.SecurityGroupIds),
+		StorageCapacity:      fileSystemOptions.StorageCapacity,
+		SubnetIds:            aws.StringSlice(fileSystemOptions.SubnetIds),
+		Tags:                 tags,
+	}
+
+	output, err := c.fsx.CreateFileSystemWithContext(ctx, &input)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == fsx.ErrCodeIncompatibleParameterError {
+				return nil, ErrAlreadyExists
+			}
+		}
+		return nil, err
+	}
+
+	return &FileSystem{
+		DnsName:         aws.StringValue(output.FileSystem.DNSName),
+		FileSystemId:    aws.StringValue(output.FileSystem.FileSystemId),
+		StorageCapacity: aws.Int64Value(output.FileSystem.StorageCapacity),
+	}, nil
+}
+
+func (c *cloud) DeleteFileSystem(ctx context.Context, fileSystemId string) error {
+	fs, err := c.getFileSystem(ctx, fileSystemId)
+	if err != nil {
+		return err
+	}
+
+	skipFinalBackup := aws.Bool(true)
+	for _, tag := range fs.Tags {
+		if *tag.Key == SkipFinalBackupTagKey {
+			if tag.Value != nil {
+				//If the tag was manually modified and is no longer a bool, set the behavior to the CSI driver default of true
+				skipFinalBackup, err = util.StringToBoolPointer(*tag.Value)
+				if err != nil {
+					skipFinalBackup = aws.Bool(true)
+				}
+			}
+			break
+		}
+	}
+
+	openZFSConfiguration := &fsx.DeleteFileSystemOpenZFSConfiguration{
```

Comment:
Shoot - another thing I missed from the design. I think it'd be good if we added a tag via the ```FinalBackupTags``` parameter to indicate that the final backup was created by the CSI driver.